File: frontend/src/api/comment.ts
export interface CommentResponse {
  id: string;
  article_id: string;
  user_id: string;
  content: string;
  created_at: string;
  reply_to_id?: string;
  replies?: CommentResponse[];
}

export interface CommentCreate {
  article_id: string;
  content: string;
  reply_to_id?: string;
}

export interface CommentUpdate {
  content?: string;
}

File: wiki-backend/app/schemas/comment.py
# app/schemas/comment.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime
from uuid import UUID

class CommentBase(BaseModel):
    content: str

class CommentCreate(CommentBase):
    article_id: UUID
    reply_to_id: Optional[UUID] = None

class CommentUpdate(BaseModel):
    content: Optional[str] = None

class CommentResponse(CommentBase):
    model_config = ConfigDict(from_attributes=True)
    
    id: UUID
    article_id: UUID
    user_id: UUID
    created_at: datetime
    reply_to_id: Optional[UUID] = None
    replies: List["CommentResponse"] = []

CommentResponse.model_rebuild()
